modules/http4s/src/smithy4s/http4s/package.scala

```diff
@@ -60,4 +60,31 @@ package object http4s extends Compat.Package {
       (CaseInsensitive(k.toString), v.map(_.value))
     }
 
+  private[smithy4s] def splitPathSegmentsAndQueryParams(
```

Comment:
Suggestion: these don't seem to depend on http4s, could we move them to smithy4s.http?

Reply:
we could , we would need to make it non package private or use it internally (i.e. in the endpoints interface and metadata)

Reply:
yeah, without @Baccata's opinion I'd keep it here then

Reply:
Right so I think the correct approach is the following : 

First, from a semantic point of view, we're gonna set that `path` is only relative to path segments, so anything with "path" in its name should not be consuming nor returning anything after the `?` in the URI. In particular, [this method](https://github.com/disneystreaming/smithy4s/blob/series/0.17/modules/core/src/smithy4s/http/internals/package.scala#L55-L63) 

So what we need is : 

1. Change the `pathSegment` implementation to just return the `path` bits, which will trickle down to the `path` method in "httpEndpoint". The `split('?')`  should therefore happen before the `split('/')`
2. add a `def staticQueryParams : Map[String, Seq[String]]` to the `HttpEndpoint`, 
3. use that method in the http4s client endpoint implementation

This will be cleaner and will be reusable by third parties